lib/reconfigure.js

```diff
@@ -79,7 +79,11 @@ module.exports = function(config) {
     for (var i = 0; i < cloudantServices.length; i++) {
       if (typeof config.instanceName === 'undefined' || cloudantServices[i].name === config.instanceName) {
         var credentials = cloudantServices[i].credentials;
-        if (credentials && credentials.url) {
+        if (credentials && credentials.apikey && credentials.username) {
+          outCreds.iamApiKey = credentials.apikey;
+          outUrl = 'https://' + encodeURIComponent(credentials.username) + '.cloudant.com';
```

Comment:
I think we should set the URL from the `host` property of the VCAP data (for both IAM and legacy auth cases). That will work better for cases where the name isn't guaranteed to be `cloudant.com`.

Reply:
Ok, sounds good, I'll update this.

Reply:
Updated in 93c3da0.